ee/tabby-db/src/migration_tests.rs

```diff
@@ -0,0 +1,51 @@
+use sqlx::migrate;
+
+use crate::DbConn;
+
+#[tokio::test]
+async fn test_repository_migration() {
+    let migrations = migrate!("./migrations");
+    let db = DbConn::new_in_memory().await.unwrap();
+
+    migrations.undo(&db.pool, 28).await.unwrap();
```

Comment:
instead of migrate then undo, is there a command like migrate to version X? this shall make the unit test easier and no need to generate `down` sql for such case.

Reply:
seems can be done with a customized wrapped MigrateSource: https://github.com/launchbadge/sqlx/blob/0449ac5c1f5220571bf8aaf0ab560d40bf603fc8/sqlx-core/src/migrate/source.rs#L25

something like MigrateSourceLessThanVersion shall do the work

Reply:
If we're going to do that, I had an easier solution before:
```rs
let migrations = migrate!("./migrations");
for migration in migrations.iter().take(28) {
  sqlx::query(&migration.sql).execute(&db.pool).await.unwrap();
}
...
let migration = migrations.iter().nth(28);
sqlx::query(&migration.sql).execute(&db.pool).await.unwrap();
```